### ISSUE TITLE
Echo: Table in Queue > Reader headers when sorted are not announced by screen reader

### ISSUE THREAD

**Issue by alisan16** [department-of-veterans-affairs/caseflow#12136]
Update on 4/24/2020: The headers in the table are not being read correctly. When you sort (e.g. ascending) it is not announced by the screen reader "Sorted ascending"

---- 

Table is styled incorrectly
https://developer.mozilla.org/en-US/docs/Learn/HTML/Tables/Advanced

Noting that this table styling (esp. sort, filter) is inconsistent with Queue tables, this could be an opportunity to tighten this up

Data displayed visually on the page beginning with the column headers: Categories, Receipt Date, Document Type, Issue Tags and Comments should be coded as a table for assistive technology users. This text content is read as paragraph text and links. Assistive technology users cannot understand the information and structure that is presented visually.  (New: 08/27/2019) the filter button would not work as well next to categories and the table is not coded as a table

**Comment by yoomlam** [comment]
@alisan16 
Unable to replicate described problems on this page:
<img width="1169" alt="image" src="https://user-images.githubusercontent.com/55255674/77080257-17631800-69c7-11ea-8502-c8d95b77d7f5.png">

> Table in Queue > Reader is not coded as a table. 

The element has `table` with `summary` attribute, `thead` with `th` columns, `tbody` with `td` cells, and `tfoot` as suggested at https://developer.mozilla.org/en-US/docs/Learn/HTML/Tables/Advanced. One possible improvement is to convert `summary` into a `caption` tag.

<img width="539" alt="ReaderTable" src="https://user-images.githubusercontent.com/55255674/77080754-ae2fd480-69c7-11ea-8516-d97e38c85aec.png">


> Filter and sort not accessible to AT

Using JAWS (and VoiceOver), I'm able to navigate to the filter and sort elements, as well as activate and use them. The popup for the filter button could be improved but that's a bigger lift as a popup is also used in QueueTable's filter buttons -- a designer's input would be useful to identify which keys should be used to close the popup and provide guidance on popup behavior (e.g., when selecting an option in the popup).

> Assistive technology users cannot understand the information and structure that is presented visually

Using the keyboard, I'm able to navigate to each column in visual order (left to right) and then to subsequent rows.  Column headers and cell values are described by JAWS correctly.

> the filter button would not work as well next to categories

I don't understand this statement. Where should the filter button be instead?

**Comment by alisan16** [comment]
There's a chance this was fixed with other a11y fixes. Let's close this and follow up with Linsey to clarify what her comments mean regarding the filter button. My guess is it's only relevant if the table was not being read as a table (original defect), so it might not apply any longer. cc @chandracarney to follow up.

**Comment by chandracarney** [comment]
@alisan16 confirmed the issue has been only partially resolved. The filter and the headers aren't working correctly. Would you like me to re-open this issue and edit it, or open a new issue entirely?

Noted by Linsey
```Partially Resolved: The table is now coded correctly as a table, however the headers are not being read correctly.  When you sort (e.g. ascending) it is not announced by the screen reader "Sorted ascending" ```

**Comment by alisan16** [comment]
Let's re-open. I'll update the summary text of the issue to clarify that the screen reader needs to announce "Sorted ascending"
